src/server/lib/oauth/callback.js

```diff
@@ -263,10 +263,18 @@ async function _getOAuthAccessToken (code, provider, callback) {
 function _get (provider, accessToken, results, callback) {
   let url = provider.profileUrl
   const headers = provider.headers || {}
+  const setGetAccessTokenProfileUrl = provider.setGetAccessTokenProfileUrl !== null ? provider.setGetAccessTokenProfileUrl : true
```

Comment:
instead of expanding the API surface, could you just check for `provider.id === "mailru"` instead?